### PR TITLE
Fix & Improve ClickableSlot

### LINKED ISSUE #9: Color codes not applying correctly when using ChatColor#applyURLs()
**Describe the bug**
ChatColor#applyURLs() creates TextComponent objects for each word in a string. This not only hurts performance, but also resets the color code for each string. This only happens when using `ChatMessage#getComponent()` to retrieve a Bukkit compatible message object.

**To Reproduce**
1. Create a ChatMessage object with color codes
2. Use `ChatColor#applyURLs()` on the ChatMessage object
3. Send component of ChatMessage object to player.

**Expected behavior**
`ChatMessage#applyURL()` should not create a TextComponent for each word in a string. It should split the message into n pieces instead, while keeping the message (and converted URLs) in the correct order.

### PR DESCRIPTION
**Pull request by kokumaji**
* Added the ability to add lore to a ClickableSlot item 
* Fixed a NPE related to the ItemMeta

### CODE REVIEW

File: src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java [hunk 1]
```diff
@@ -4,40 +4,55 @@
...
-    
-    @Getter
-    private ItemStack item;
-
-    @Getter
-    private int slot;
 
-    @Getter
-    private ItemMeta meta = item.getItemMeta();
+    @Getter public ItemStack item;
```
Comment (vladfrangu) [inline src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java]: Please keep this on separate lines

File: src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java [hunk 2]
```diff
@@ -44,18 +46,30 @@ public static URLPair findURL(@NotNull String text) {
      *             formatted URLs.
      * @return {@link TextComponent}
      */
-    public static TextComponent convertURLs(@NotNull String text) {
+
+    public static TextComponent convertURLs(String text) {
```
Comment (vladfrangu) [inline src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java]: Changes unrelated to this PR..

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java [hunk 3]
```diff
@@ -4,40 +4,55 @@
...
+    @Getter public int slot;
 
-    public ClickableSlot(Material pMaterial, String pName, int pAmount, int pSlot) {
+    public ClickableSlot(Material pMaterial, String pName, int pAmount, int pSlot, String... lore) {
```
Comment (ZachyFoxx) [inline src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java]: Use `NotNull` and `Nullable` annotations for parameters please.

File: src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java [hunk 4]
```diff
@@ -4,40 +4,55 @@
...
+            metaLore.add(loreComments);
+        }
+
+        isM.setLore(metaLore);
         item.setItemMeta(isM);
 
         this.slot = pSlot;
     }
 
+    public ClickableSlot(ItemStack item, int pSlot) {
```
Comment (ZachyFoxx) [inline src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java]: Parameter annotations please

File: src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java (hunk 2 above)
Comment (ZachyFoxx) [inline src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java]: Please use param annotations!

Also, this is unrelated to this PR, however I personally don't mind as long as you list it somewhere.

**Review by ZachyFoxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java [hunk 5]
```diff
@@ -4,41 +4,82 @@
...
+    /**
+     * Creates a ClickableSlot object that should be used
+     * to register interactive slots in a {@link GUI} object
+     * @param pMaterial The Material for this slot item
+     * @param pName The Displayname for this slot item
+     * @param pAmount Amount of items
+     * @param pSlot The slot which should be used for this item
+     * @param lore (Optional) Add lore to this item
+     */
+    public ClickableSlot(@NotNull Material pMaterial, String pName, int pAmount, int pSlot, String... lore) {
```
Comment (vladfrangu) [inline src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java]: You should reverse the name and amount, and mark the amount as NotNull as well

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java [hunk 6]
```diff
@@ -4,41 +4,82 @@
...
 
-    public void execute(Runnable r) {
-        new Thread(r).start();
+    /**
+     * Creates a ClickableSlot object that should be used
+     * to register interactive slots in a {@link GUI} object
+     * @param item A pre-configured {@link org.bukkit.inventory.ItemStack} object
+     * @param pSlot The slot which should be used for this item
+     */
+    public ClickableSlot(@NotNull ItemStack item, int pSlot) {
```
Comment (vladfrangu) [inline src/main/java/com/dumbdogdiner/stickyapi/bukkit/gui/ClickableSlot.java]: Missed one <3

File: src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java [hunk 7]
```diff
@@ -44,18 +46,30 @@ public static URLPair findURL(@NotNull String text) {
...
-        for (String s : text.split(" ")) {
+        TextComponent tmp = new TextComponent();
+
+        String[] split = text.split(" ");
+        int i = 0;
+        for(String s : split) {
             URLPair url = findURL(s + " ");
-            if ((url) == null) {
-                finalComp.addExtra(s + " ");
+            if((url) == null) {
```
Comment (vladfrangu) [inline src/main/java/com/dumbdogdiner/stickyapi/common/util/url/URLUtil.java:58]:
```suggestion
            if(url == null) {
```

**Review by ZachyFoxx — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**